In [2]:
import pandas as pd 

# 관광객 2535개
visitor = pd.read_excel("주요관광지점 입장객-수정.xlsx")
# 카테고리 10731개 
category = pd.read_csv("lod_category.csv",encoding='utf-8')

In [5]:
visitor= visitor[visitor['내/외국인'] == '합계']
visitor

시도    군구               관광지 내/외국인  2019년 01월  2019년 02월  2019년 03월  \
1         강원도   춘천시  KT&G 상상마당 춘천아트센터    합계    17807.0    19239.0    22139.0   
4         강원도   춘천시           강원도립화목원    합계     3735.0     4147.0     6738.0   
7         강원도   춘천시         강원도청소년수련관    합계     2685.0     1631.0     2624.0   
10        강원도   춘천시            강원숲체험장    합계     1315.0     1190.0     1097.0   
13        강원도   춘천시            강촌레일파크    합계    18818.0    27522.0    39690.0   
...       ...   ...               ...   ...        ...        ...        ...   
6376  제주특별자치도  서귀포시             천제연폭포    합계    32685.0    35115.0    41862.0   
6379  제주특별자치도  서귀포시             천지연폭포    합계    79290.0    86019.0   101259.0   
6382  제주특별자치도  서귀포시             카멜리아힐    합계    70533.0    58711.0    58207.0   
6385  제주특별자치도  서귀포시             퍼시픽랜드    합계    44161.0    49058.0    46542.0   
6388  제주특별자치도  서귀포시        한화아쿠아플라넷제주    합계    59601.0    64673.0    55539.0   

      2019년 04월  2019년 05월  2019년 06월  ...  2019년 09월  2019년 10월  2019년 11월  \
1           NaN        NaN        NaN  ...    25501.0    36257.0    22429.0   
4       19896.0    32512.0    16331.0  ...     9037.0    36460.0    15634.0   
7        3365.0     5038.0     4033.0  ...     2284.0     3696.0     1685.0   
10       1587.0     2039.0     2278.0  ...     1787.0     2344.0     1950.0   
13      68862.0    78026.0    63606.0  ...    41962.0    78484.0    51418.0   
...         ...        ...        ...  ...        ...        ...        ...   
6376    52541.0    48699.0    50957.0  ...    43270.0    62051.0    43215.0   
6379   128873.0   142454.0   113574.0  ...   100782.0   140915.0   108609.0   
6382    63505.0    58358.0    65266.0  ...    42035.0    53919.0    65407.0   
6385    65376.0    89972.0    68582.0  ...    70687.0        NaN        NaN   
6388    66411.0    79535.0    73261.0  ...    76753.0    82900.0    54752.0   

      2019년 12월  2020년 01월  2020년 02월  2020년 03월  2020년 04월  2020년 05월  \
1       21066.0    18727.0    10913.0    13660.0        NaN        NaN   
4       34524.0     4128.0     2298.0     6833.0        NaN        NaN   
7        1713.0     2731.0      861.0       54.0        NaN        NaN   
10        842.0     1328.0     1271.0     1135.0        NaN        NaN   
13      32702.0    26248.0    14476.0    11284.0        NaN        NaN   
...         ...        ...        ...        ...        ...        ...   
6376    34852.0        NaN        NaN        NaN        NaN        NaN   
6379    80254.0        NaN        NaN        NaN        NaN        NaN   
6382    91152.0        NaN        NaN        NaN        NaN        NaN   
6385        NaN        NaN        NaN        NaN        NaN        NaN   
6388    66155.0        NaN        NaN        NaN        NaN        NaN   

      2020년 06월  
1           NaN  
4           NaN  
7           NaN  
10          NaN  
13          NaN  
...         ...  
6376        NaN  
6379        NaN  
6382        NaN  
6385        NaN  
6388        NaN  

[2535 rows x 22 columns]

In [7]:
category.columns = ['관광지','카테고리']
category

관광지       카테고리
0              구서오시게시장       상설시장
1             부산금정문화회관        공연장
2               이바구공작소        전시관
3           장기려기념더나눔센터        기념관
4                168계단       이색거리
...                ...        ...
10726            멍딩이마을  농.산.어촌 체험
10727  교동민화마을관광두레주민사업체       이색거리
10728             제천향교    유적지/사적지
10729            복천사제천         사찰
10730            미타사음성         사찰

[10731 rows x 2 columns]

In [8]:
# 그냥 관광지명으로 merge하면? 564개 
pd.merge(category, visitor, on='관광지')

관광지       카테고리     시도   군구 내/외국인  2019년 01월  2019년 02월  2019년 03월  \
0      국립부산과학관        전시관  부산광역시  기장군    합계   102316.0    88808.0    84181.0   
1         아홉산숲    자연생태관광지  부산광역시  기장군    합계     9730.0     8453.0     9425.0   
2       삼탄아트마인       이색체험    강원도  정선군    합계     1526.0     1837.0     1482.0   
3      국립산악박물관        박물관    강원도  속초시    합계     7881.0     7420.0     5946.0   
4     설악산자생식물원        수목원    강원도  속초시    합계     1056.0     1424.0     3337.0   
..         ...        ...    ...  ...   ...        ...        ...        ...   
559     상수허브랜드       관광농원   충청북도  청주시    합계     4926.0     5909.0     9204.0   
560       도담삼봉          강   충청북도  단양군    합계   163854.0   219358.0   308758.0   
561  괴산한지체험박물관        박물관   충청북도  괴산군    합계      453.0      371.0      440.0   
562        반야사         사찰   충청북도  영동군    합계     5692.0     6935.0     5548.0   
563      한두레권역  농.산.어촌 체험   충청북도  옥천군    합계      330.0      118.0      341.0   

     2019년 04월  2019년 05월  ...  2019년 09월  2019년 10월  2019년 11월  2019년 12월  \
0     102529.0    94529.0  ...    66063.0    96597.0    66025.0    70179.0   
1      10221.0    13711.0  ...    10211.0    12071.0    11204.0     6112.0   
2       1768.0     2922.0  ...     2792.0     4020.0     2520.0     2164.0   
3       4574.0     9441.0  ...     6103.0    12441.0     7222.0    11557.0   
4       4576.0    10869.0  ...     6304.0     8485.0     4503.0     2104.0   
..         ...        ...  ...        ...        ...        ...        ...   
559     9451.0    11291.0  ...     5104.0     5251.0     3592.0     3039.0   
560   527517.0   571217.0  ...   378386.0   566333.0   439938.0   219573.0   
561      699.0      540.0  ...      547.0     1177.0     1198.0      575.0   
562     7746.0    12715.0  ...     9884.0    12526.0    10738.0     7260.0   
563      920.0     1118.0  ...      584.0     1234.0      952.0      240.0   

     2020년 01월  2020년 02월  2020년 03월  2020년 04월  2020년 05월  2020년 06월  
0      73209.0    14509.0        0.0        0.0     4833.0    11573.0  
1       1482.0     2330.0     5049.0     6011.0    17806.0     7800.0  
2          NaN        NaN        NaN        NaN        NaN        NaN  
3          NaN        NaN        NaN        NaN        NaN        NaN  
4          NaN        NaN        NaN        NaN        NaN        NaN  
..         ...        ...        ...        ...        ...        ...  
559      754.0        0.0        0.0        0.0        0.0        0.0  
560        NaN        NaN        NaN        NaN        NaN        NaN  
561      302.0      222.0        0.0        NaN        NaN        NaN  
562        NaN        NaN        NaN        NaN        NaN        NaN  
563        NaN        NaN        NaN        NaN        NaN        NaN  

[564 rows x 23 columns]

In [12]:
# str.contains로 하면? 우선, dictionary만들고, key-value 받아서? 
# 1. category 사전만들기 
category_dic = dict(zip(list(category['관광지']), list(category['카테고리'])))
category_dic['강문해변'] # 관광지를 넣으면 category가 뽑힌다아

'해수욕장'

In [38]:
site_name = list(category_dic.keys())
aa = site_name[1] # 관광지 이름을 받음 
category_dic[aa] #관광지 이름에 해당하는 카테고리 유형 받음 

'공연장'

In [72]:
# 2. str.contains(카테고리에 있는 장소들) 
site_name = list(category_dic.keys())
visitor2 = visitor.copy()
visitor2['카테고리'] = np.nan
for i in range(len(site_name)) :
    temp_index = visitor2[visitor2['관광지'].str.contains(site_name[i])].index  # 이름 포함된 행 index 가져오기 
    visitor2.loc[temp_index,'카테고리'] =  category_dic[site_name[i]]  # 카테고리 컬럼에 카테고리 넣기 
visitor2    

시도    군구               관광지 내/외국인  2019년 01월  2019년 02월  2019년 03월  \
1         강원도   춘천시  KT&G 상상마당 춘천아트센터    합계    17807.0    19239.0    22139.0   
4         강원도   춘천시           강원도립화목원    합계     3735.0     4147.0     6738.0   
7         강원도   춘천시         강원도청소년수련관    합계     2685.0     1631.0     2624.0   
10        강원도   춘천시            강원숲체험장    합계     1315.0     1190.0     1097.0   
13        강원도   춘천시            강촌레일파크    합계    18818.0    27522.0    39690.0   
...       ...   ...               ...   ...        ...        ...        ...   
6376  제주특별자치도  서귀포시             천제연폭포    합계    32685.0    35115.0    41862.0   
6379  제주특별자치도  서귀포시             천지연폭포    합계    79290.0    86019.0   101259.0   
6382  제주특별자치도  서귀포시             카멜리아힐    합계    70533.0    58711.0    58207.0   
6385  제주특별자치도  서귀포시             퍼시픽랜드    합계    44161.0    49058.0    46542.0   
6388  제주특별자치도  서귀포시        한화아쿠아플라넷제주    합계    59601.0    64673.0    55539.0   

      2019년 04월  2019년 05월  2019년 06월  ...  2019년 10월  2019년 11월  2019년 12월  \
1           NaN        NaN        NaN  ...    36257.0    22429.0    21066.0   
4       19896.0    32512.0    16331.0  ...    36460.0    15634.0    34524.0   
7        3365.0     5038.0     4033.0  ...     3696.0     1685.0     1713.0   
10       1587.0     2039.0     2278.0  ...     2344.0     1950.0      842.0   
13      68862.0    78026.0    63606.0  ...    78484.0    51418.0    32702.0   
...         ...        ...        ...  ...        ...        ...        ...   
6376    52541.0    48699.0    50957.0  ...    62051.0    43215.0    34852.0   
6379   128873.0   142454.0   113574.0  ...   140915.0   108609.0    80254.0   
6382    63505.0    58358.0    65266.0  ...    53919.0    65407.0    91152.0   
6385    65376.0    89972.0    68582.0  ...        NaN        NaN        NaN   
6388    66411.0    79535.0    73261.0  ...    82900.0    54752.0    66155.0   

      2020년 01월  2020년 02월  2020년 03월  2020년 04월  2020년 05월  2020년 06월    카테고리  
1       18727.0    10913.0    13660.0        NaN        NaN        NaN     NaN  
4        4128.0     2298.0     6833.0        NaN        NaN        NaN     NaN  
7        2731.0      861.0       54.0        NaN        NaN        NaN     NaN  
10       1328.0     1271.0     1135.0        NaN        NaN        NaN     NaN  
13      26248.0    14476.0    11284.0        NaN        NaN        NaN  복합 레포츠  
...         ...        ...        ...        ...        ...        ...     ...  
6376        NaN        NaN        NaN        NaN        NaN        NaN      폭포  
6379        NaN        NaN        NaN        NaN        NaN        NaN     NaN  
6382        NaN        NaN        NaN        NaN        NaN        NaN     수목원  
6385        NaN        NaN        NaN        NaN        NaN        NaN    테마공원  
6388        NaN        NaN        NaN        NaN        NaN        NaN     전시관  

[2535 rows x 23 columns]

In [76]:
visitor2[visitor2['카테고리'].notnull()]

시도    군구         관광지 내/외국인  2019년 01월  2019년 02월  2019년 03월  \
13        강원도   춘천시      강촌레일파크    합계    18818.0    27522.0    39690.0   
16        강원도   춘천시        구곡폭포    합계     7118.0     6309.0     8287.0   
27        강원도   춘천시      남이섬유원지    합계   140434.0   148032.0   180164.0   
65        강원도   춘천시  엘리시안강촌-골프장    합계        0.0     2696.0     9353.0   
68        강원도   춘천시  엘리시안강촌-스키장    합계    93413.0    78162.0     2450.0   
...       ...   ...         ...   ...        ...        ...        ...   
6370  제주특별자치도  서귀포시       제주추사관    합계     7470.0     6179.0     6551.0   
6376  제주특별자치도  서귀포시       천제연폭포    합계    32685.0    35115.0    41862.0   
6382  제주특별자치도  서귀포시       카멜리아힐    합계    70533.0    58711.0    58207.0   
6385  제주특별자치도  서귀포시       퍼시픽랜드    합계    44161.0    49058.0    46542.0   
6388  제주특별자치도  서귀포시  한화아쿠아플라넷제주    합계    59601.0    64673.0    55539.0   

      2019년 04월  2019년 05월  2019년 06월  ...  2019년 10월  2019년 11월  2019년 12월  \
13      68862.0    78026.0    63606.0  ...    78484.0    51418.0    32702.0   
16      15009.0    21071.0    16578.0  ...    20666.0    14623.0     8394.0   
27     298687.0   280562.0   244649.0  ...   404302.0   333178.0   192197.0   
65      12131.0    12947.0    13000.0  ...    10954.0     9577.0     3407.0   
68          0.0        0.0        0.0  ...        0.0     2622.0    48950.0   
...         ...        ...        ...  ...        ...        ...        ...   
6370     9680.0    10938.0     6819.0  ...     8311.0     7388.0     5323.0   
6376    52541.0    48699.0    50957.0  ...    62051.0    43215.0    34852.0   
6382    63505.0    58358.0    65266.0  ...    53919.0    65407.0    91152.0   
6385    65376.0    89972.0    68582.0  ...        NaN        NaN        NaN   
6388    66411.0    79535.0    73261.0  ...    82900.0    54752.0    66155.0   

      2020년 01월  2020년 02월  2020년 03월  2020년 04월  2020년 05월  2020년 06월    카테고리  
13      26248.0    14476.0    11284.0        NaN        NaN        NaN  복합 레포츠  
16      11974.0     8768.0    19335.0        NaN        NaN        NaN      폭포  
27     143987.0    71873.0    44487.0        NaN        NaN        NaN    관광단지  
65       2385.0     3282.0     8997.0        NaN        NaN        NaN    관광단지  
68      74866.0    27218.0      160.0        NaN        NaN        NaN    관광단지  
...         ...        ...        ...        ...        ...        ...     ...  
6370        NaN        NaN        NaN        NaN        NaN        NaN     전시관  
6376        NaN        NaN        NaN        NaN        NaN        NaN      폭포  
6382        NaN        NaN        NaN        NaN        NaN        NaN     수목원  
6385        NaN        NaN        NaN        NaN        NaN        NaN    테마공원  
6388        NaN        NaN        NaN        NaN        NaN        NaN     전시관  

[874 rows x 23 columns]

## str.contains로 해서 관광지 포함된 관광지에 일괄적으로 관광지 카테고리 값 넣었음! 
### 그냥 merge하면 564개 카테고리만 넣을 수 있는데 str.contains하면 874개 카테고리 넣을 수 있음 

In [64]:
# 점봉산 / 점봉산곰배령

visitor2[visitor2['관광지'].str.contains(site_name[555])].loc[:,'카테고리'] = category_dic[site_name[555]]

C:\Users\wewe8\anaconda3\lib\site-packages\pandas\core\indexing.py:966: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item] = s


In [71]:
temp_index = visitor2[visitor2['관광지'].str.contains(site_name[555])].index
visitor2.loc[temp_index,'카테고리'] =  category_dic[site_name[555]]
visitor2.loc[temp_index,'카테고리']

678    산
Name: 카테고리, dtype: object

In [77]:
visitor2[visitor2['관광지'].str.contains(site_name[555])]

시도   군구     관광지 내/외국인  2019년 01월  2019년 02월  2019년 03월  2019년 04월  \
678  강원도  인제군  점봉산곰배령    합계      731.0      521.0        0.0     3268.0   

     2019년 05월  2019년 06월  ...  2019년 10월  2019년 11월  2019년 12월  2020년 01월  \
678    15803.0    13903.0  ...     3268.0    15803.0    13903.0        NaN   

     2020년 02월  2020년 03월  2020년 04월  2020년 05월  2020년 06월  카테고리  
678        NaN        NaN        NaN        NaN        NaN     산  

[1 rows x 23 columns]